examples/addr2line.rs

```diff
@@ -0,0 +1,208 @@
+extern crate gimli;
+extern crate getopts;
+
+use std::env;
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let mut opts = getopts::Options::new();
+    opts.optopt("e", "exe", "Set the input file name (default is a.out)", "<executable>");
+
+    let matches = opts.parse(&args[1..]).unwrap();
+    let file_path = matches.opt_str("e").unwrap_or("a.out".to_string());
+    let file = obj::open(&file_path);
+    if obj::is_little_endian(&file) {
+        symbolicate::<gimli::LittleEndian>(&file, &matches);
+    } else {
+        symbolicate::<gimli::BigEndian>(&file, &matches);
+    }
+}
+
+fn parse_uint_from_hex_string(string: &str) -> u64
+{
+    if string.len() > 2 {
+        if string.starts_with("0x") {
+            return u64::from_str_radix(&string[2..], 16).expect("Failed to parse address");
+        }
+    }
+
+    return u64::from_str_radix(string, 16).expect("Failed to parse address");
```

Comment:
Nit: rust style is to leave out `return` and the semicolon when in tail position.

It's not just functions either. Rust's block "statements" will evaluate to their last expression, if you drop the semicolon, eg:

```rust
let foo = {
    bar();
    1
};
assert_eq(foo, 1);
```

Same for `if`s, etc.